common/autoware_ad_api_msgs/planning/msg/SteeringFactor.msg

```diff
@@ -0,0 +1,29 @@
+# constants for common use
+uint16 UNKNOWN = 0
+
+# constants for type
+uint16 INTERSECTION = 1
+uint16 LANE_CHANGE = 2
+uint16 AVOIDANCE_PATH_CHANGE = 3
+uint16 AVOIDANCE_PATH_RETURN = 4
+uint16 STATION = 5
+uint16 PULL_OUT = 6
+uint16 PULL_OVER = 7
+uint16 EMERGENCY_OPERATION = 8
+
+# constants for direction
+uint16 LEFT = 1
+uint16 RIGHT = 2
+
+# constants for status
+uint16 APPROACHING = 1
+uint16 TRYING = 2
+uint16 TURNING = 3
+
+# variables
+geometry_msgs/Pose[] pose
+float32[] distance
```

Comment:
How about changing this to a fixed length array? For intersections, assign the same value.

Reply:
@isamu-takagi Do you mean fix by using this way?
```
geometry_msgs/Pose[2] pose
float32[2] distance
```

Reply:
Yes. I think this is easier for users to use because it does not need to check length.

Reply:
fixed the array to 2 in this [8675cc4](https://github.com/autowarefoundation/autoware.universe/pull/1706/commits/8675cc47a8ae71b7e825b4cf7e97fa5fab029454)

Reply:
fix the vector error in [21339f8](https://github.com/autowarefoundation/autoware.universe/pull/1706/commits/21339f8175f39bf0ce5da8562c120f3d214d112d)